radix-engine/src/model/resource.rs

```diff
@@ -0,0 +1,282 @@
+use sbor::*;
+use scrypto::engine::types::*;
+use scrypto::rust::collections::BTreeMap;
+use scrypto::rust::collections::BTreeSet;
+use scrypto::rust::collections::HashMap;
+use scrypto::rust::rc::Rc;
+use scrypto::rust::string::ToString;
+
+/// Represents some error when manipulating resources.
+#[derive(Debug, Clone)]
+pub enum ResourceError {
+    /// Resource addresses do not match
+    ResourceAddressNotMatching,
+    /// The amount is invalid, according to the resource divisibility
+    InvalidAmount(Decimal, u8),
+    /// The balance is not enough
+    InsufficientBalance,
+    /// The operation is not supported
+    UnsupportedOperation,
+}
+
+/// Represents the contained resource.
+#[derive(Debug, Clone, TypeId, Encode, Decode)]
+pub enum Resource {
```

Comment:
Perhaps description should be something like: "Represents the locked/liquid state of the contained resource". Perhaps also change the name from `Resource` to `ResourceContainerState`. `Resource` is a bit too generic.

Reply:
Like the proposed name. Will change